### PR TITLE
Add tests for whatwg.org redirects

### PR DESCRIPTION
**Pull request by foolip**
This also makes run-tests.sh do nothing outside of the master branch,
and keeps running tests even on failures.

### CODE REVIEW

**Comment by annevk** [comment]
Looks great. Might also be worth testing that `http://whatwg.org/C` first redirects to `https://whatwg.org/C`.

**Comment by foolip** [comment]
Things are actually a bit broken for http://whatwg.org, see this line:
```
  ['http://whatwg.org/', 301, 'http://www.whatwg.org/', 'keep'], // TODO
```

With these tests in place, I'm finalizing the nginx hosting of whatwg.org, and then I'll be able to fix most of the TODOs.

**Comment by foolip** [comment]
So, with that explanation, I am going to boldly merge this so that I can see it work on Travis and then upload the whatwg.org hosting.

File: run-tests.sh [hunk 1]
```diff
@@ -3,9 +3,28 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+# do nothing on Travis except for the master branch
+TRAVIS=${TRAVIS:-false}
+if [[ "$TRAVIS" == "true" && "$TRAVIS_BRANCH" != "master" ]]; then
```
Comment (domenic) [inline run-tests.sh]: Can't you use .travis.yml's `branches` option?

File: run-tests.sh [hunk 2]
```diff
@@ -3,9 +3,28 @@ set -o errexit
...
+    echo "Skipping tests for branch $TRAVIS_BRANCH (not master)"
+    exit 0
+fi
+
 cd tests
+npm install node-fetch
```
Comment (domenic) [inline run-tests.sh]: A `package.json` is the more conventional way to do this.

File: run-tests.sh (hunk 1 above)
Comment (foolip) [reply run-tests.sh]: https://github.com/whatwg/misc-server/pull/21

File: run-tests.sh (hunk 2 above)
Comment (foolip) [reply run-tests.sh]: Like https://github.com/whatwg/misc-server/pull/22?

**Pull request merged**